packages/math/least_common_multiple.pony

```diff
@@ -0,0 +1,6 @@
+primitive LeastCommonMultiple
+  """
+  Get the least common multiple of x and y
+  """
+  fun apply(x: USize, y: USize): USize ? =>
```

Comment:
Why `USize` specifically?

It would probably be "best" to write these as generics that take a type parameter instead of hard-coding a particular integer type.

But if we have to hard-code a particular integer type, it would likely be better to not use a platform-dependent type.

Also, does this algorithm you've used in the two functions only work for unsigned integers, or can it also work for negative integers as well?

Reply:
Generics is doable.

GCD works for negative.
LCM isn't technically allowed for negatives, you are supposed to discard the signs so for example LCM(-2,-6) is the same as LCM(2,6) and you calculate accordingly.

For LCM we can either only allow unsigned or I can update the implementation to return the negation of the result if it is less than 0.

Reply:
There's also a question with least common multiple about what we do for overflow given that the answer could be more than the highest value given. Perhaps use partial arithmetic?